gqlopencensus/metrics.go

```
@@ -0,0 +1,201 @@
+	view.Unregister(resolverCompletedCounterView)
+	view.Unregister(timeToResolveFieldView)
+	view.Unregister(timeToHandleRequestView)
+}
+
+func ResolverMiddleware() graphql.FieldMiddleware {
+	return func(ctx context.Context, next graphql.Resolver) (interface{}, error) {
+		rctx := graphql.GetResolverContext(ctx)
+
+		ctx, err := tag.New(context.Background(),
+			tag.Insert(keyObject, rctx.Object),
+			tag.Insert(keyField, rctx.Field.Name),
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		stats.Record(ctx, resolverStartedCounter.M(1))
+		// resolverStartedCounter.WithLabelValues(rctx.Object, rctx.Field.Name).Inc()
+
+		observerStart := time.Now()
+
+		res, err := next(ctx)
+
+		var exitStatus string
+		if err != nil {
+			exitStatus = existStatusFailure
+		} else {
+			exitStatus = exitStatusSuccess
+		}
+
+		ctx, err = tag.New(context.Background(),
+			tag.Insert(keyExitStatus, exitStatus),
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		stats.Record(ctx, timeToResolveField.M(time.Since(observerStart).Nanoseconds()/int64(time.Millisecond)))
+		//timeToResolveField.With(prometheusclient.Labels{"exitStatus": exitStatus}).
+		//	Observe(float64(time.Since(observerStart).Nanoseconds() / int64(time.Millisecond)))
+
+		stats.Record(ctx, resolverCompletedCounter.M(1))
+		// resolverCompletedCounter.WithLabelValues(rctx.Object, rctx.Field.Name).Inc()
+
+		return res, err
+	}
+}
+
+func RequestMiddleware() graphql.RequestMiddleware {
+	return func(ctx context.Context, next func(ctx context.Context) []byte) []byte {
+		// requestStartedCounter.Inc()
+		stats.Record(ctx, requestStartedCounter.M(1))
+
+		observerStart := time.Now()
+
+		res := next(ctx)
+
+		rctx := graphql.GetResolverContext(ctx)
+		reqCtx := graphql.GetRequestContext(ctx)
+		errList := reqCtx.GetErrors(rctx)
+
+		var exitStatus string
+		if len(errList) > 0 {
+			exitStatus = existStatusFailure
+		} else {
+			exitStatus = exitStatusSuccess
+		}
+
+		ctx, _ = tag.New(context.Background(),
```

seems like I have to ignore this error or panic?